### ISSUE TITLE
Static BIN file format

### ISSUE THREAD

**Issue by oleomachado** [opensourceBIM/BIMsurfer#116]
I found that the static implementation do something that I want to do.

```
   displayStaticBuffer("glcanvas", "duplex.bin");

  export function displayStaticBuffer(domNode, bufferPath) {
      var canvas = document.getElementById(domNode);
      let v = new BimServerViewer({quantizeVertices:false}, canvas, window.innerWidth, window.innerHeight, null);
      v.loadAnnotationsFromPreparedBufferUrl(bufferPath);
  }
```

Is there any documentation of the .BIN format?

How could I generate it for another IFC file?

**Comment by aothms** [comment]
`.bin` is probably the binary buffer of a glTF v1 file. These days everybody is using glTF v2 `.glb` where both the hierarchical info (json) and binary geometry info is combined into a single binary file (typically). With IfcConvert you can generate .glb files from .ifc.

**Comment by oleomachado** [comment]
I checked, but the available BIN doesn't seem to be a Gltf Bin V1 or V2 (no magic number, not following the specification).

But I believe it was created using this repo https://github.com/opensourceBIM/BinarySerializers.

I will continue my research :)

Thanks

**Comment by hlg** [comment]
Yes, this was probably created from the binary serializer plugins. The format is not documented besides through the serializer implementation code itself.

To create these files, you just need to run a BIMserver instance (local or remote) with the BinarySerializer plugins installed, upload your IFC to it and download the binary file. You can do the upload and download through a web client like BIMvie.ws or any other client using the BIMserver API.
